docs/content/concepts/assets/external-assets.mdx

```diff
@@ -0,0 +1,193 @@
+from dagster import AssetSpec, Definitions, external_assets_from_specs
+
+asset_one = AssetSpec("asset_one")
+asset_two = AssetSpec("asset_two", deps=[asset_one])
+
+defs = Definitions(assets=external_assets_from_specs([asset_one, asset_two]))
+```
+
+## Keeping an external asset's metadata up to date.
+
+Since with external assets Dagster does not control scheduling or materialization, it is up to the user to keep the metadata for an external asset up-to-date. There are a number of mechanisms for doing so:
+
+- REST APIs
+- Sensors
+- Log events in ops
+
+### Using REST API
+
+Dagster OSS exposes a REST endpoint for reporting asset materializations.
+
+Here's an example `curl` command in a shell script.
+
+```bash
+curl --request POST \
+    --url https://path/to/instance/report_asset_materialization/{asset_key}\
+    --header 'Content-Type: application/json' \
+    --data '{
+    "metadata" : {
+        "source": "From curl command
+    }
+}'
+```
+
+Here's an example of invoking this API in Python with `requests`:
+
+```python
+import requests
+
+url = f"https://path/to/instance/report_asset_materialization/{asset_key}"
+payload = { "metadata": { "source": "From python script" } }
+headers = { "Content-Type": "application/json" }
+
+response = requests.request("POST", url, json=payload, headers=headers)
+```
+
+There are also end points for reporting asset observations and asset check evaluations.
+
+### Using Sensors
+
+You can generate events to attach to external assets and provide them to sensors directly, by using the `asset_events` param of <PyObject object="SensorResult"/>.
+
+```python file=/concepts/assets/external_assets/external_asset_using_sensor.py
+import datetime
+
+from dagster import (
+    AssetMaterialization,
+    AssetSpec,
+    Definitions,
+    SensorEvaluationContext,
+    SensorResult,
+    external_asset_from_spec,
+    sensor,
+)
+
+
+def utc_now_str() -> str:
+    return datetime.datetime.now(datetime.timezone.utc).strftime("%Y-%m-%d, %H:%M:%S")
+
+
+@sensor()
+def keep_external_asset_a_up_to_date(context: SensorEvaluationContext) -> SensorResult:
+    # Materialization happened in external system, but is recorded here
+    return SensorResult(
+        asset_events=[
+            AssetMaterialization(
+                asset_key="external_asset_a",
+                metadata={
+                    "source": f'From sensor "{context.sensor_name}" at UTC time "{utc_now_str()}"'
+                },
+            )
+        ]
+    )
+
+
+defs = Definitions(
+    assets=[external_asset_from_spec(AssetSpec("external_asset_a"))],
+    sensors=[keep_external_asset_a_up_to_date],
+)
+```
+
+### Using the Python API
+
+Users can insert events to attach to external assets directly from Dagster's Python API. An example where this would be useful is writing a handrolled Python script to backfill metadata. The relevant API is `report_runless_asset_event` on <PyObject object="DagsterInstance"/>.
```

Comment:
```suggestion
You can insert events to attach to external assets directly from Dagster's Python API. An example where this would be useful is writing a hand-rolled Python script to backfill metadata. The relevant API is `report_runless_asset_event` on <PyObject object="DagsterInstance"/>.
```